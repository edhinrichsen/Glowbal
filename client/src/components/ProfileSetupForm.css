.profile-form {
  width: 500px;
  padding: 20px 60px 60px 60px;
}

.ant-form-item .ant-input {
  border-radius: var(--border-radius-primary) !important;
}

.ant-picker {
  border-radius: var(--border-radius-primary) !important;
}

.ant-picker-panel-container {
  border-radius: var(--border-radius-primary) !important;
}
