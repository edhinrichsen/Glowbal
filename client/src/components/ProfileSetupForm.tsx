import React from 'react';

import { Button, Form, Input, Select, Switch } from 'antd';

import Emoji from './Emoji';

import './ProfileSetupForm.css';

const { Option } = Select;

type ProfileSetupFormProps = {
  /** Callback function when the form is submitted */
  onFinish?: (result: any) => void;
  /** Text to be included on the submit button */
  submitLabel?: string;
  /** Used to display loading icon on submit button */
  isLoading?: boolean;
};

function ProfileSetupForm(props: ProfileSetupFormProps) {
  const onFinish = (result: any) => {
    if (props.onFinish) {
      props.onFinish(result);
    }
  };

  return (
    <div className="profile-form">
      <Form
        name="basic"
        initialValues={{
          remember: true,
          public: true,
          lookingForWork: true,
          publicLocation: true,
        }} // public and lookingForWork are set due to inconsistent inital value of Switch
        onFinish={onFinish}
      >
        <h3>
          <strong>Personal Information</strong>
        </h3>

        <Form.Item
          label="First Name"
          name="firstName"
          rules={[
            {
              required: true,
              message: 'Please enter your first name',
              max: 50,
            },
          ]}
        >
          <Input />
        </Form.Item>

        <Form.Item
          label="Last Name"
          name="lastName"
          rules={[
            {
              required: true,
              message: 'Please enter your last name',
              max: 50,
            },
          ]}
        >
          <Input />
        </Form.Item>

        <Form.Item
          label="Gender"
          name="gender"
          rules={[
            {
              required: true,
              message: 'Please enter your gender or select prefer not to say',
            },
          ]}
        >
          <Select>
            <Option value="male">
              <Emoji symbol="👨‍💼" />
              Male
            </Option>
            <Option value="female">
              <Emoji symbol="👩‍💼" />
              Female
            </Option>
            <Option value="other">
              <Emoji symbol="🧑‍💼" />
              Other / Prefer not to say
            </Option>
          </Select>
        </Form.Item>

        {/* TODO: DOB: currently returns type as Moment which API does not accept (requires JS Date) - Can't see a way for DatePicker to convert from Moment to Date
        
        <Form.Item label="Date of Birth" name="DOB" rules={[]}>
          <DatePicker
            picker="date"
            format="DD/MM/YYYY"
            placeholder="DD/MM/YYYY"
          />
        </Form.Item> */}

        <h3>
          <strong>Profile Information</strong>
        </h3>

        <Form.Item
          label="Profile Privacy"
          name="public"
          valuePropName="checked"
        >
          <Switch
            checkedChildren={
              <>
                <Emoji symbol="🌏" /> Public
              </>
            }
            unCheckedChildren={
              <>
                <Emoji symbol="⛔️" /> Private
              </>
            }
          />
        </Form.Item>

        <Form.Item
          label="Show your location"
          name="publicLocation"
          valuePropName="checked"
        >
          <Switch
            checkedChildren={
              <>
                <Emoji symbol="🌏" /> Public
              </>
            }
            unCheckedChildren={
              <>
                <Emoji symbol="⛔️" /> Private
              </>
            }
          />
        </Form.Item>

        <Form.Item
          label="Work Status"
          name="lookingForWork"
          valuePropName="checked"
        >
          <Switch
            checkedChildren={
              <>
                <Emoji symbol="✅" /> Looking for work
              </>
            }
            unCheckedChildren={
              <>
                <Emoji symbol="❌" /> Not looking for work
              </>
            }
          />
        </Form.Item>

        <Form.Item>
          <Button
            type="primary"
            htmlType="submit"
            loading={props.isLoading ? props.isLoading : false}
          >
            {props.submitLabel ? props.submitLabel : 'Submit'}
          </Button>
        </Form.Item>
      </Form>
    </div>
  );
}

export default ProfileSetupForm;
