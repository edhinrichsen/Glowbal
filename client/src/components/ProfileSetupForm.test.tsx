import React from 'react';
import { shallow } from 'enzyme';
import ProfileSetupForm from './ProfileSetupForm';

describe('Testing ProfileSetupForm', () => {
  it('renders without crashing', () => {
    const component = shallow(<ProfileSetupForm />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
