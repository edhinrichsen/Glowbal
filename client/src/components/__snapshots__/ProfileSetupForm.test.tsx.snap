// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProfileSetupForm renders without crashing 1`] = `
<div
  className="profile-form"
>
  <ForwardRef(InternalForm)
    initialValues={
      Object {
        "lookingForWork": true,
        "public": true,
        "publicLocation": true,
        "remember": true,
      }
    }
    name="basic"
    onFinish={[Function]}
  >
    <h3>
      <strong>
        Personal Information
      </strong>
    </h3>
    <FormItem
      label="First Name"
      name="firstName"
      rules={
        Array [
          Object {
            "max": 50,
            "message": "Please enter your first name",
            "required": true,
          },
        ]
      }
    >
      <Input
        type="text"
      />
    </FormItem>
    <FormItem
      label="Last Name"
      name="lastName"
      rules={
        Array [
          Object {
            "max": 50,
            "message": "Please enter your last name",
            "required": true,
          },
        ]
      }
    >
      <Input
        type="text"
      />
    </FormItem>
    <FormItem
      label="Gender"
      name="gender"
      rules={
        Array [
          Object {
            "message": "Please enter your gender or select prefer not to say",
            "required": true,
          },
        ]
      }
    >
      <Select
        bordered={true}
        choiceTransitionName=""
        transitionName="slide-up"
      >
        <Option
          value="male"
        >
          <Emoji
            symbol="👨‍💼"
          />
          Male
        </Option>
        <Option
          value="female"
        >
          <Emoji
            symbol="👩‍💼"
          />
          Female
        </Option>
        <Option
          value="other"
        >
          <Emoji
            symbol="🧑‍💼"
          />
          Other / Prefer not to say
        </Option>
      </Select>
    </FormItem>
    <h3>
      <strong>
        Profile Information
      </strong>
    </h3>
    <FormItem
      label="Profile Privacy"
      name="public"
      valuePropName="checked"
    >
      <Switch
        checkedChildren={
          <React.Fragment>
            <Emoji
              symbol="🌏"
            />
             Public
          </React.Fragment>
        }
        unCheckedChildren={
          <React.Fragment>
            <Emoji
              symbol="⛔️"
            />
             Private
          </React.Fragment>
        }
      />
    </FormItem>
    <FormItem
      label="Show your location"
      name="publicLocation"
      valuePropName="checked"
    >
      <Switch
        checkedChildren={
          <React.Fragment>
            <Emoji
              symbol="🌏"
            />
             Public
          </React.Fragment>
        }
        unCheckedChildren={
          <React.Fragment>
            <Emoji
              symbol="⛔️"
            />
             Private
          </React.Fragment>
        }
      />
    </FormItem>
    <FormItem
      label="Work Status"
      name="lookingForWork"
      valuePropName="checked"
    >
      <Switch
        checkedChildren={
          <React.Fragment>
            <Emoji
              symbol="✅"
            />
             Looking for work
          </React.Fragment>
        }
        unCheckedChildren={
          <React.Fragment>
            <Emoji
              symbol="❌"
            />
             Not looking for work
          </React.Fragment>
        }
      />
    </FormItem>
    <FormItem>
      <Button
        block={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        type="primary"
      >
        Submit
      </Button>
    </FormItem>
  </ForwardRef(InternalForm)>
</div>
`;
